api/src/main/java/org/openmrs/module/fhir/api/strategies/observation/ObservationStrategy.java

```diff
@@ -216,6 +219,53 @@ public Observation updateFHITObservation(Observation observation, String uuid) {
 				createObservation(observation, uuid);
 	}
 
+	/**
+	 * @see GenericObservationStrategy#searchObservationByPatientAndCode(java.lang.String, java.util.List)
+	 */
+	@Override
+	public List<Observation> searchObservationByPatientAndCode(String patientUuid, List<TokenParam> codings) {
+		Patient patient = Context.getPatientService().getPatientByUuid(patientUuid);
+		List<Observation> obsList = new ArrayList<>();
+		if (codings.isEmpty()) {
+			return obsList;
+		} else {
+			for (TokenParam tokenParam : codings) {
+				obsList.addAll(getObservations(patient, getConcept(tokenParam)));
+			}
+		}
+		return new ArrayList<Observation>(new HashSet<>(obsList));
```

Comment:
Why not just return the obsList?

Reply:
Remove duplicates observations

Reply:
How do they come up?

Reply:
If you search using` PIH|5086, LOINC|53482-6`, You will get duplicate results since the two refer to the same concept.

Reply:
If the client decides to use duplicate concepts, i would let them receive such, instead of complicating the code. 😊

Reply:
ok